lib/python/ensembl/brc4/runnable/dump_stable_ids.py

```diff
@@ -0,0 +1,468 @@
+
+        """
+        query = """SELECT old_stable_id, new_stable_id
+        FROM stable_id_event
+        WHERE (old_stable_id != new_stable_id OR old_stable_id IS NULL OR new_stable_id IS NULL)
+            AND type="gene"
+            AND mapping_session_id=%s
+        """
+        values = (session_id,)
+        cursor = self.server.get_cursor()
+        cursor.execute(query, values)
+
+        pairs = []
+        for db in cursor:
+            pair = {"old_id": db[0], "new_id": db[1]}
+            pairs.append(pair)
+        print(f"{len(pairs)} stable id events")
+        return pairs
+    
+    def make_events(self, pairs: List) -> List:
+        """Given a list of pairs, create events.
+
+        Args:
+            pairs: list of dicts {'old_id': str, 'new_id': str}.
+        
+        Return:
+            A list of events.
+
+        """
+
+        from_list = {}
+        to_list = {}
+        for pair in pairs:
+            old_id = pair["old_id"]
+            new_id = pair["new_id"]
+            if old_id == None:
+                old_id = ""
+            if new_id == None:
+                new_id = ""
+
+            if old_id in from_list:
+                from_list[old_id].add(new_id)
+            else:
+                from_list[old_id] = set(new_id)
+
+            if new_id in to_list:
+                to_list[new_id].add(old_id)
+            else:
+                to_list[new_id] = set(old_id)
+        
+        # Remove empty elements
+        for from_id in from_list:
+            from_list[from_id] = StableIdEvent.clean_set(from_list[from_id])
+        for to_id in to_list:
+            to_list[to_id] = StableIdEvent.clean_set(to_list[to_id])
+        
+        events = []
+        for old_id in from_list:
+            if not old_id or old_id not in from_list: continue
+            event = StableIdEvent([old_id], from_list[old_id])
+            (event, from_list, to_list) = self.extend_event(event, from_list, to_list)
+            event.add_pairs(pairs)
+            events.append(event)
+        
+        # Remaining events should only be new genes
+        for new_id in to_list:
+            if not new_id: continue
+            event = StableIdEvent(to_list[new_id], [new_id])
+            event.add_pairs(pairs)
+            events.append(event)
+        
+        stats = {}
+        for event in events:
+            name = event.get_name()
+            if not name in stats:
+                stats[name] = 1
+            else:
+                stats[name] += 1
+        
+        for stat in stats:
+            print(f"\t{stat} = {stats[stat]}")
+
+        return events
+    
+    def extend_event(self, event: StableIdEvent, from_list: Dict[str, List[str]], to_list: Dict[str, List[str]]
+        ) -> Tuple[StableIdEvent, List, List]:
+        """Given an event, add ids in to from and to sets, as long as they are connected either in
+        a from_list or a to_list.
```

Comment:
Just a comment (no need to change it): the first line of the docstring should be a short description that should fit in the line length.

Reply:
I did shorten it, I had a doctring commit to do anyway...